app/src/main/java/com/sunasterisk/travelapp/ui/base/BaseMVPFragment.kt

```diff
@@ -1,18 +1,28 @@
 package com.sunasterisk.travelapp.ui.base
 
 import android.app.Dialog
+import android.os.Bundle
 import android.view.View
-import androidx.fragment.app.Fragment
 import com.google.android.material.snackbar.Snackbar
 import com.sunasterisk.travelapp.R
 import com.sunasterisk.travelapp.utils.createProgressDialog
 import com.sunasterisk.travelapp.utils.showSnackBar
 import com.sunasterisk.travelapp.utils.showToast
 
-abstract class BaseMVPFragment<T> : Fragment(), BaseContract.View<T> {
 
+abstract class BaseMVPFragment<V : BaseContract.View<T>, T : BaseContract.Presenter<V>> :
+    BaseFragment(),
+    BaseContract.View<T> {
+
+    protected abstract val presenter: T
     private var progressDialog: Dialog? = null
 
+    @Suppress("UNCHECKED_CAST")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        presenter.onAttach(this as V)
```

Comment:
check cast trước khi cast chứ 
```suggestion
        if(this is V){
            presenter.onAttack(this)
        }
```

Reply:
vâng ạ

Reply:
thế thì kiểu cast nullable có được k 
```suggestion
        (this as? V)?.let { presenter.onAttach(it) }
```

Reply:
vâng được ạ